executor/executor.go

```diff
@@ -105,7 +106,7 @@ func (e *Executor) createWorker() {
 
 // Run executes [f] after all previously enqueued [f] with
 // overlapping [conflicts] are executed.
-func (e *Executor) Run(conflicts set.Set[string], f func() error) {
+func (e *Executor) Run(conflicts state.Keys, f func() error) {
```

Comment:
Can you add a "TODO" with a link to an issue to handle read-only/write-only keys (everything rn is just treated as RW)?